cpp/tests/unary/unary_ops_test.cu

```diff
@@ -55,6 +55,155 @@ cudf::test::fixed_width_column_wrapper<T> create_expected_columns(cudf::size_typ
+
+    cudf::test::fixed_width_column_wrapper<T> input    (std::cbegin(h_input_v),  std::cend(h_input_v));
+    cudf::test::fixed_width_column_wrapper<T> expected (std::cbegin(h_expect_v), std::cend(h_expect_v));
+
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::SQRT);
+
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+TYPED_TEST(cudf_math_test, SimpleABS)
+{
+    cudf::test::fixed_width_column_wrapper<TypeParam> input    {{ -2, -1, 1, 2 }};
+    cudf::test::fixed_width_column_wrapper<TypeParam> expected {{  2,  1, 1, 2 }};
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::ABS);
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+TYPED_TEST(cudf_math_test, SimpleSQRT)
+{
+    cudf::test::fixed_width_column_wrapper<TypeParam> input    {{ 1, 4, 9, 16 }};
+    cudf::test::fixed_width_column_wrapper<TypeParam> expected {{ 1, 2, 3, 4  }};
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::SQRT);
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+template <typename T>
+struct cudf_math_with_floating_point_test : public cudf::test::BaseFixture {};
+
+using floating_point_type_list = ::testing::Types<float, double>;
+
+TYPED_TEST_CASE(cudf_math_with_floating_point_test, floating_point_type_list);
+
+TYPED_TEST(cudf_math_with_floating_point_test, SimpleSIN)
+{
+    cudf::test::fixed_width_column_wrapper<TypeParam> input    {{ 0.0 }};
+    cudf::test::fixed_width_column_wrapper<TypeParam> expected {{ 0.0 }};
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::SIN);
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+TYPED_TEST(cudf_math_with_floating_point_test, SimpleCOS)
+{
+    cudf::test::fixed_width_column_wrapper<TypeParam> input    {{ 0.0 }};
+    cudf::test::fixed_width_column_wrapper<TypeParam> expected {{ 1.0 }};
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::COS);
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+TYPED_TEST(cudf_math_with_floating_point_test, SimpleFLOOR)
+{
+    cudf::test::fixed_width_column_wrapper<TypeParam> input    {{ 1.1, 3.3, 5.5, 7.7 }};
+    cudf::test::fixed_width_column_wrapper<TypeParam> expected {{ 1.0, 3.0, 5.0, 7.0 }};
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::FLOOR);
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
+TYPED_TEST(cudf_math_with_floating_point_test, SimpleCEIL)
+{
+    cudf::test::fixed_width_column_wrapper<TypeParam> input    {{ 1.1, 3.3, 5.5, 7.7 }};
+    cudf::test::fixed_width_column_wrapper<TypeParam> expected {{ 2.0, 4.0, 6.0, 8.0 }};
+    auto output = cudf::experimental::unary_operation(input, cudf::experimental::unary_op::CEIL);
+    cudf::test::expect_columns_equal(expected, output->view());
+}
+
```

Comment:
I think we need 
* test of unary ops on error cases (which things should throw exceptions?)
* test of unary ops on empty inputs (result should be empty outputs of appropriate type).

Reply:
`normalized_unary_op` refactor done (and I very good suggestion I might add - I am sad I didn't realize this myself).

Reply:
failures / throwing tests added as well